README.md

```diff
@@ -174,3 +174,93 @@ Then create a new release:
 5. Once the NPM package has been published you will need to update the dependency to
 it in both the `server` and `app` repositories so that they both refer to the
 latest contract ABI when talking to the blockchain.
+
+## Business logic and smart contract structure
+
+### Actors
+
+- Event organser = the creator of the event contract. There is only one per event.
+- Event admins = the people who can help the event organiser. Can do everything the organiser can do apart from transfering the contract ownership and destroying the event contract deployer contract.
+- Participants = People who did RSVP the event
+- Attendees = People who were marked as attended
+
+### Event flow
+
+- When the event organiser creates an event, the organiser can specify the unit of currency (ETH or ERC20), name of the event(`name`), amount of commitment (`deposit`), event capaicty (`limitOfParticipants`), and cooling period (`cooling period`).
+- The event organiser adds admins
+- Until the first participant RSVP, the event admins can change the name and `deposit` amount
+- The vent admins can change limitOfParticipants to increase the capacity of the event.
+- When participants RSVP, they can RSVP if they call `register` function with `deposit`.
+- For events participants commit ERC20 tokens, the user has to call `token.approve(deposit)` prior to RSVP.
+- When user arrives to the venue, admins can check in users. This will be done off chain.
+- If the event is canceled, each participant can withdraw the deposit amount (minus gas fee)
+- Once the event is over, one of the admins call `finalize` sending the array of participants information in the bitmap format (to save gas).
+- Once the event is finalised, the attendees can withdraw the proportion of the sum of the deposit on the contract. For example, if the `deposit` is 0.1 ETH , 10 people RSVP, and 5 people turned up, then each attendee can withdraw 1/5 = 0.2 ETH.
+- If the attendees do not withdraw within cooling period, one of the admins can `clear` so that the remaining `deposit` will be send to the event organiser.
+
+### Contract files
+
+```
+~/.../kickback/contracts (spike-dai)$tree contracts/
+contracts/
+├── AbstractConference.sol
+├── Conference.sol
+├── Deployer.sol
+├── DeployerInterface.sol
+├── ERC20Conference.sol
+├── ERC20Deployer.sol
+├── EthConference.sol
+├── EthDeployer.sol
+├── GroupAdmin.sol
+├── Migrations.sol
+├── MyToken.sol
+└── zeppelin
+    ├── lifecycle
+    │   └── Destructible.sol
+    └── ownership
+        └── Ownable.sol
+
+```
+
+
+- Each even has its own contract. `AbstractConference.sol` is the superclass which contains all the functions. 
+- The `AbstractConference.sol` inherits from Distructible, Ownnable and GroupAdmin.
+- `ERC20Conference.sol` and `EthConference.sol` inherits `AbstractConference.sol` and override logic to RSVP and withdraw.
+- `Deployer.sol` is a factory contract to deploy smart contract for each event.
```

Comment:
Maybe we should add which 'smart contract' you are referring to to make it clear, or show a list of the ones it's supposed to deploy (or a link to a list)